### DISCUSSION TITLE
[Feature] Request for Feature: Sync Photos from Server to Mobile Client（Device）

### DISCUSSION THREAD

**Discussion by zhaolin-st** [immich-app/immich#11344] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature Sync Photos from Server to Mobile Client

Hello,

First of all, thank you for developing and maintaining Immich. It has been incredibly useful for managing and backing up my photos.

I have a use case that I'd like to share. I have successfully uploaded all my photos from my mobile devices and computer to the Immich server. However, I do not have a public IP, so I am unable to access my album remotely at all times.

I am wondering if it would be possible to implement a feature on the mobile client that allows synchronization of photos from the server to the mobile device. As far as I understand, the current functionality supports uploading photos from the mobile device to the server, but there is no feature to sync photos from the server back to the mobile device (similar to how iCloud Photos allows you to keep original files on the device).

Is there any plan to introduce such a feature in future releases? It would greatly enhance the usability by allowing users to access their photos anytime, even without constant internet access.

Thank you for considering this feature request.

Best regards,
Beer

### Platform

- [X] Server
- [ ] Web
- [X] Mobile

**Comment by kaygdotorg** [comment]
> However, I do not have a public IP, so I am unable to access my album remotely at all times.

I assume you meant a public static IP. Plenty of users have this scenario, and use something like Cloudflare Tunnel to access their immich instances remotely.

**Reply by zhaolin-st** [reply]
Thank you for the suggestion. I understand that using a public static IP or solutions like Cloudflare Tunnel can facilitate remote access to my Immich instance. However, my request is about enhancing the functionality of the Immich mobile client to support synchronization of photos from the server to the mobile device.

Having a synchronization feature would allow my mobile device to store a complete copy of my photo library, enabling offline access and use of photos without relying on a specific app or continuous internet connection. This functionality would be similar to how iCloud Photos allows users to keep original files on their devices.

While external access solutions are helpful, they don't fully address the need for local data availability on mobile devices. I believe this feature would greatly improve user experience by providing seamless and consistent access to personal photo libraries.

Thank you for considering this.

**Reply by kaygdotorg** [reply]
Yeah, I understand your request is about something else. I too would love something like that, as previously with Nextcloud, I used to keep my pictures synchronised with my devices.

I just wanted to point out that you can still get remote access to your immich instance without a public IP. :)

**Comment by bo0tzz** [comment]
#5695

**Comment by rrakso** [comment]
I'm forcing myself to switch from `Synology Photos` to `Immich`, but this "two way sync" feature is crucial for me :c
<img width="300" alt="image" src="https://github.com/user-attachments/assets/039ab8c6-ff5e-4073-8409-fcbdfa40e73b" />
This is how they've implemeted that solution.
